File: frontend/src/services/moodService.ts
import { api } from "./api";

export interface MoodEntry {
  id: number;
  user_id: number;
  mood: string;
  date: string;
}

export const moodHistory = async (): Promise<MoodEntry[]> => (await api.get("/mood/history")).data;
export const logMood = async (mood: string, date: string): Promise<MoodEntry> =>
  (await api.post("/mood", { mood, date })).data;

File: backend/app/services/mood_service.py
from sqlalchemy.orm import Session

from app.models.mood import MoodEntry
from app.schemas.mood import MoodCreate


def log_mood(db: Session, user_id: int, couple_id: int, payload: MoodCreate) -> MoodEntry:
    entry = MoodEntry(user_id=user_id, couple_id=couple_id, **payload.model_dump())
    db.add(entry)
    db.commit()
    db.refresh(entry)
    return entry


def history(db: Session, couple_id: int, limit: int = 14) -> list[MoodEntry]:
    return (
        db.query(MoodEntry)
        .filter(MoodEntry.couple_id == couple_id)
        .order_by(MoodEntry.id.desc())
        .limit(limit)
        .all()
    )
